### ISSUE TITLE
How to get weights of features in Logistic Regression Model?

### ISSUE THREAD

**Issue by Sheryl-Wang** [facebookresearch/CrypTen#286]
Hi, I try to train a logistic regression model to do binary classification, with 4 input features. Since the explanation in tutorial7 said that the label in encrypt model training must be one-hot encoded, the linear in my logistic reg is (input=4, output=2). 
I use one-label label in the training, and use BCELoss, and finally get the output. I understand for each sample, the class with larger output can ne used as the predicted class. But I wonder how can I get the weights of feature and bias of the final model, because I need them for future work such as credit scoring.

More explanation:
Like in sklearn logsitic reg, the one-hot label is not required, so we can get one set of coeffi and intercept of the final model. But in crypten, we get two sets of weights and bias in my case, how can I get the similar parameters as in sklearn for my next credit scoring.

Many thanks for your kind help!

**Comment by Sheryl-Wang** [comment]
Add the code
x.size: torch.Size([100, 4])
y.size: torch.Size([100, 2])


```
x_enc = crypten.cryptensor(x)
eye = torch.eye(2)
y_train = eye[y]
y_train_enc = crypten.cryptensor(y_train)
dummy_input = torch.empty((1, 4))

layer_linear = nn.Linear(4, 2)
layer_enc = crypten.nn.from_pytorch(layer_linear, dummy_input)
layer_enc.encrypt()
layer_enc.train()

loss = crypten.nn.BCELoss()
learning_rate = 0.001

result_enc = layer_enc(x_enc)
sig_enc = result_enc.sigmoid()
loss_value = loss(sig_enc, y_train_enc)
print_loss = loss_value.get_plain_text().item()

layer_enc.zero_grad()
loss_value.backward()
layer_enc.update_parameters(learning_rate) 

for name, parameter in layer_enc.decrypt().named_parameters():
  print(name, parameter)

```


The result:
![image](https://user-images.githubusercontent.com/64888952/125886302-366db17e-28dc-4cec-8338-4ca68177e31f.png)

**Comment by knottb** [comment]
For `BCELoss`, you shouldn't need one-hot encoding since the output is a single probability rather than a set of probabilities. The one-hot encoding should only be for `CrossEntropyLoss`.

I think you are getting the parameters properly. `layer_enc.decrypt()` should decrypt all of the parameters of the CrypTen model and can be accessed using `layer_enc.decrypt().parameters()` or `layer_enc.decrypt().named_parameters()`.

Is there something more you are looking for here?
